crates/reward/src/lib.rs

```diff
@@ -424,6 +426,10 @@ where
 {
     type CurrencyId = T::CurrencyId;
 
+    fn reward_currencies_len(pool_id: &T::PoolId) -> u32 {
+        RewardCurrencies::<T, I>::get(pool_id).len() as u32
```

Comment:
This would cause a storage read before the extrinsic is included. Would that be exploitable somehow?

Reply:
Got an answer, seems we need to provide an explicit upper bound for the number of iterations and check that at runtime.